scala/scala.bzl

```diff
@@ -494,6 +511,55 @@ def _scala_test_impl(ctx):
     _write_test_launcher(ctx, rjars)
     return _scala_binary_common(ctx, cjars, rjars)
 
+def _prep_grep_pattern_from(patterns):
+    combined_pattern = ""
+    for pattern in patterns:
+      combined_pattern += " -e {}\.class".format(pattern)
+    return combined_pattern
+
+def _discover_classes(ctx, patterns, archive):
+    discovered_classes = ctx.new_file(ctx.label.name + "_discovered_classes.txt")
+    ctx.action(
+      inputs=[archive],
+      outputs=[discovered_classes],
+      progress_message="Discovering classes with patterns of %s" % patterns,
+      #TODO consider with Damien/Ulf/Oscar the implications of switching from grep to scala code
+      #Pro-> logic will be cohesive (currently the scala code assumes stuff from the grep)
+      #Con-> IIRC Ulf warned me about performance implications of these traversals
+      command="jar -tf {archive} | grep {combined_patterns} > {out}".format(archive=archive.path, combined_patterns=_prep_grep_pattern_from(patterns), out=discovered_classes.path))
```

Comment:
Thanks for the reference! I indeed tried it before using the shell `jar` but what I missed was the need for `_jdk` which the [comment](https://github.com/bazelbuild/rules_scala/blob/master/thrift/thrift.bzl#L67) on line 67 gave me.
It's fixed but while fixing it I noticed that `_implicit_deps` contain an entry for `_jar` but it points to `"//src/java/io/bazel/rulesscala/jar:binary_deploy.jar"`. This sounds to me like it should be `jar_creator_deploy_jar` or something of the sort. WDYT?